src/riak_kv_vnode.erl

```diff
@@ -2704,6 +2697,45 @@ highest_actor(ActorBase, Obj) ->
     %% get the greatest event for the highest/latest actor
     {Actor, Epoch, riak_object:actor_counter(Actor, Obj)}.
 
+%%
+%% Technical note:  The index_module configuration parameter should contain
+%% a module name which must implement the following functions:
+%%
+%%     - index(object_pair(), write_reason(), p()) -> ok.
+%%     - index_binary(bucket(), key(), binary(), write_reason(), p()) -> ok.
+%%     - is_searchable(riak_kv_bucket:props()) -> boolean().
+%%
```

Comment:
Would it be feasible to define these functions as callbacks in a behavior module? If so, the `yokozuna.erl` module (and any other future indexing modules that we might create) could use the behavior to get some extra compile-time checking.